dotcom-rendering/src/web/layouts/CommentLayout.tsx

```diff
@@ -698,48 +699,59 @@ export const CommentLayout = ({ CAPIArticle, NAV, format }: Props) => {
 					/>
 				</ElementContainer>
 
-				{CAPIArticle.storyPackage && (
-					<ElementContainer>
-						<Island deferUntil="visible">
-							<Carousel
-								heading={CAPIArticle.storyPackage.heading}
-								trails={CAPIArticle.storyPackage.trails.map(
-									decideTrail,
-								)}
-								onwardsType="more-on-this-story"
+				{CAPIArticle.onwards ? (
+					<DecideOnwards
+						onwards={CAPIArticle.onwards}
+						format={format}
+					/>
+				) : (
```

Comment:
Would it be cleaner to pass the CAPIArticle to <DecideOnwards> and do all of the possible onwards rendering in that component? It seems like we have onwards rendering logic both places now.

Reply:
(Applies to all layouts)

Reply:
There's a rule in DCR to never pass `CAPIArticle` as a prop. We have this because this object is many thousands of lines long and can reach upwards of 30Kbs in size. It's preferable to keep props as focussed and explicit as possible to make the work React has to do easier and to keep the components easy to reason about.